pkg/k8s/node_info.go

```diff
@@ -0,0 +1,36 @@
+package k8s
+
+import v1 "k8s.io/api/core/v1"
+
+// NodeInfo provides an abstraction on top of node to pods mappings
+// replaces scheduler cache.NodeInfo that was removed from public in an older version of kubernetes
+// Maintains the same interface
+type NodeInfo struct {
+	node *v1.Node
+	pods []*v1.Pod
+}
+
+// NewNodeInfo creates a new empty NodeInfo struct
+func NewNodeInfo() *NodeInfo {
+	return &NodeInfo{}
+}
+
+// AddPod adds a pod to the list of pods for this node
+func (i *NodeInfo) AddPod(pod *v1.Pod) {
+	i.pods = append(i.pods, pod)
```

Comment:
Does this need a lock? What happens if we have two routines add a pod at the same time?

Reply:
Will add a RWLock